packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx

```diff
@@ -63,11 +145,46 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
         [SocialAddressType.NEXT_ID],
     )
 
+    const { value: currentVisitingSocialIdentity, loading: loadingCurrentVisitingSocialIdentity } =
+        useCurrentVisitingSocialIdentity()
+
+    const { value: personaProof, retry: retryProof } = useAsyncRetry(async () => {
+        if (!currentVisitingSocialIdentity?.publicKey) return
+        return NextIDProof.queryExistedBindingByPersona(currentVisitingSocialIdentity?.publicKey)
+    }, [currentVisitingSocialIdentity?.publicKey])
+
+    useEffect(() => {
+        return MaskMessages.events.ownProofChanged.on(() => {
+            retryProof()
+        })
+    }, [retryProof])
+
+    const wallets = personaProof?.proofs?.filter((proof) => proof?.platform === NextIDPlatform.Ethereum)
+
+    const addressList = useMemo(() => {
+        if (!wallets?.length || (!isOwnerIdentity && socialAddressList?.length)) {
+            setSelectedAddress(first(socialAddressList))
+            return socialAddressList
+        }
+        const addresses = wallets.map((proof) => {
```

Comment:
The `useSocialAddressListAll` should already fetch addresses from NextID. Why it's still necessary to query it again?